hoodie-client/src/main/java/com/uber/hoodie/index/HoodieIndex.java

```diff
@@ -68,22 +67,24 @@ protected HoodieIndex(HoodieWriteConfig config, JavaSparkContext jsc) {
    * file
    */
   public abstract JavaPairRDD<HoodieKey, Optional<String>> fetchRecordLocation(
-      JavaRDD<HoodieKey> hoodieKeys, final HoodieTable<T> table);
+      JavaRDD<HoodieKey> hoodieKeys, final JavaSparkContext jsc,
+      HoodieTableMetaClient metaClient);
 
   /**
    * Looks up the index and tags each incoming record with a location of a file that contains the
    * row (if it is actually present)
    */
   public abstract JavaRDD<HoodieRecord<T>> tagLocation(JavaRDD<HoodieRecord<T>> recordRDD,
-      HoodieTable<T> hoodieTable) throws HoodieIndexException;
+      JavaSparkContext jsc, HoodieTableMetaClient metaClient) throws HoodieIndexException;
```

Comment:
I think its better to pass the HoodieTable here? It houses the FileSystemView which can be directly used by the index.. To support cached filesystem view, we need to end up making that change anyway. Do you see any problems with that?

Reply:
Had responded to a similar comment earlier. Essentially, with index moved inside hoodie table, passing the table down to the index operations sounds counter-intuitive. Also, left a comment below on thoughts around moving hoodie metadata operations into metaclient rather that table (for eg. timeline). Thoughts ?